swap/src/main.rs

```diff
@@ -30,95 +37,97 @@ mod trace;
-pub const PORT: u16 = 9876; // Arbitrarily chosen.
-pub const ADDR: &str = "127.0.0.1";
-pub const BITCOIND_JSON_RPC_URL: &str = "http://127.0.0.1:8332";
-
-#[cfg(feature = "tor")]
-pub const TOR_PORT: u16 = PORT + 1;
-
 #[tokio::main]
 async fn main() -> Result<()> {
-    let opt = Options::from_args();
+    let opt: Options = Options::from_args();
 
     trace::init_tracing(LevelFilter::Debug)?;
 
-    #[cfg(feature = "tor")]
-    let (addr, _ac) = {
-        let tor_secret_key = torut::onion::TorSecretKeyV3::generate();
-        let onion_address = tor_secret_key
-            .public()
-            .get_onion_address()
-            .get_address_without_dot_onion();
-        (
-            format!("/onion3/{}:{}", onion_address, TOR_PORT),
-            create_tor_service(tor_secret_key).await?,
-        )
-    };
-    #[cfg(not(feature = "tor"))]
-    let addr = format!("/ip4/{}/tcp/{}", ADDR, PORT);
-
-    let alice: Multiaddr = addr.parse().expect("failed to parse Alice's address");
-
-    if opt.as_alice {
-        info!("running swap node as Alice ...");
-
-        if opt.piconeros.is_some() || opt.satoshis.is_some() {
-            bail!("Alice cannot set the amount to swap via the cli");
+    match opt {
+        Options::Alice {
+            bitcoind_url: url,
+            listen_addr,
+            tor_port,
+            ..
+        } => {
+            info!("running swap node as Alice ...");
+
+            let behaviour = Alice::default();
+            let local_key_pair = behaviour.identity();
+
+            let (listen_addr, _ac, transport) = match tor_port {
+                Some(tor_port) => {
+                    let tor_secret_key = torut::onion::TorSecretKeyV3::generate();
+                    let onion_address = tor_secret_key
+                        .public()
+                        .get_onion_address()
+                        .get_address_without_dot_onion();
+                    let onion_address_string = format!("/onion3/{}:{}", onion_address, tor_port);
+                    let addr: Multiaddr = onion_address_string.parse()?;
+                    let ac = create_tor_service(tor_secret_key, tor_port).await?;
+                    let transport = build_tor(local_key_pair, addr.clone(), tor_port)?;
+                    (addr, Some(ac), transport)
+                }
+                None => {
+                    let transport = build(local_key_pair)?;
+                    (listen_addr, None, transport)
+                }
+            };
+
+            let bitcoin_wallet = Wallet::new("alice", &url)
+                .await
+                .expect("failed to create bitcoin wallet");
+
+            let redeem = bitcoin_wallet
+                .new_address()
+                .await
+                .expect("failed to get new redeem address");
+            let punish = bitcoin_wallet
+                .new_address()
+                .await
+                .expect("failed to get new punish address");
```

Comment:
Once you rebase address generation should be inside of the `{alice, bob}::swap` functions. Just an FYI.